JotunnLib/Managers/ZoneManager.cs

```diff
@@ -1,47 +1,300 @@
+                    continue;
+                }
+
+                biomes.Add(area);
+            }
+            return biomes;
+        }
+
+#pragma warning restore S3265 // Non-flags enums should not be used in bitwise operations
+
+        /// <summary>
+        ///     Get a ZoneLocation by its name.<br /><br />
+        ///     Search hierarchy:
+        ///     <list type="number">
+        ///         <item>Custom Location with the exact name</item>
+        ///         <item>Vanilla Location with the exact name from <see cref="ZoneSystem"/></item>
+        ///     </list>
+        /// </summary>
+        /// <param name="name">Name of the ZoneLocation to search for.</param>
+        /// <returns>The existing ZoneLocation, or null if none exists with given name</returns>
+        public ZoneLocation GetZoneLocation(string name)
+        {
+            if (Locations.TryGetValue(name, out CustomLocation customLocation))
+            {
+                return customLocation.ZoneLocation;
+            }
+
+            if (ZoneSystem.instance.m_locationsByHash.TryGetValue(name.GetStableHashCode(), out ZoneLocation location))
+            {
+                return location;
+            }
+            return null;
+        }
+
+        /// <summary>
+        ///     Create an empty GameObject that is disabled, so any Components in instantiated GameObjects will not start their lifecycle.
+        /// </summary>
+        /// <param name="name">Name of the location</param>
+        /// <returns></returns>
+        public GameObject CreateLocationContainer(string name)
+        {
+            GameObject container = new GameObject()
+            {
+                name = name
+            };
+            container.transform.SetParent(LocationContainer.transform);
+            return container;
+        }
+
+        private readonly Regex copyRegex = new Regex(@" \([0-9]+\)");
+
+        /// <summary>
+        ///     Create a copy that is disabled, so any Components in instantiated GameObjects will not start their lifecycle     
+        /// </summary>
+        /// <param name="gameObject">Prefab to copy</param>
+        /// <param name="fixLocationReferences">Replace JVLmock GameObjects with a copy of their real prefab</param>
+        /// <returns></returns>
+        public GameObject CreateLocationContainer(GameObject gameObject, bool fixLocationReferences = false)
+        {
+            var locationContainer = Object.Instantiate(gameObject, LocationContainer.transform);
+            if (fixLocationReferences)
+            {
+                var transform = locationContainer.transform;
+                FixMockReferences(transform);
+            }
             
-            Logger.LogInfo("Registering custom zone data");
+            return locationContainer;
+        }
 
-            // Call event handlers to load prefabs
-            OnBeforeZoneLoad?.Invoke(null, EventArgs.Empty);
+        private void FixMockReferences(Transform transform)
```

Comment:
Why did you write your own mocking code? GameObject.FixReferences() would have been fine, also traversing the child GOs

Reply:
I couldn't find where complete GameObjects were replaced? I would need the same "copyRegex" there, to allow multiple copies of the same mock (or you'd need to do a lot of nesting workarounds)

Reply:
Moved to FixReference extension method